presto-main/src/main/java/com/facebook/presto/operator/ExplainAnalyzeOperator.java

```diff
@@ -0,0 +1,145 @@
+            this.metadata = requireNonNull(metadata, "metadata is null");
+        }
+
+        @Override
+        public List<Type> getTypes()
+        {
+            return ImmutableList.of(VARCHAR);
+        }
+
+        @Override
+        public Operator createOperator(DriverContext driverContext)
+        {
+            checkState(!closed, "Factory is already closed");
+            OperatorContext operatorContext = driverContext.addOperatorContext(operatorId, planNodeId, ExplainAnalyzeOperator.class.getSimpleName());
+            return new ExplainAnalyzeOperator(operatorContext, queryPerformanceFetcher, metadata);
+        }
+
+        @Override
+        public void close()
+        {
+            closed = true;
+        }
+
+        @Override
+        public OperatorFactory duplicate()
+        {
+            return new ExplainAnalyzeOperatorFactory(operatorId, planNodeId, queryPerformanceFetcher, metadata);
+        }
+    }
+
+    private final OperatorContext operatorContext;
+    private final QueryPerformanceFetcher queryPerformanceFetcher;
+    private final Metadata metadata;
+    private boolean finishing;
+    private boolean outputConsumed;
+
+    public ExplainAnalyzeOperator(OperatorContext operatorContext, QueryPerformanceFetcher queryPerformanceFetcher, Metadata metadata)
+    {
+        this.operatorContext = requireNonNull(operatorContext, "operatorContext is null");
+        this.queryPerformanceFetcher = requireNonNull(queryPerformanceFetcher, "queryPerformanceFetcher is null");
+        this.metadata = requireNonNull(metadata, "metadata is null");
+    }
+
+    @Override
+    public OperatorContext getOperatorContext()
+    {
+        return operatorContext;
+    }
+
+    @Override
+    public List<Type> getTypes()
+    {
+        return ImmutableList.of(VARCHAR);
+    }
+
+    @Override
+    public void finish()
+    {
+        finishing = true;
+    }
+
+    @Override
+    public boolean isFinished()
+    {
+        return finishing && outputConsumed;
+    }
+
+    @Override
+    public boolean needsInput()
+    {
+        return !finishing;
+    }
+
+    @Override
+    public void addInput(Page page)
+    {
+        checkState(needsInput());
+
+        // Ignore the input
+    }
+
+    @Override
+    public Page getOutput()
+    {
+        if (!finishing) {
+            return null;
+        }
+        outputConsumed = true;
+        QueryInfo queryInfo = queryPerformanceFetcher.getQueryInfo(operatorContext.getDriverContext().getTaskId().getQueryId());
+        // Skip the output stage, since that's the one that has the ExplainAnalyzeOperator itself
+        String plan = textDistributedPlan(queryInfo.getOutputStage().getSubStages(), metadata, operatorContext.getSession());
+        BlockBuilder builder = VARCHAR.createBlockBuilder(new BlockBuilderStatus(), 1);
+        VARCHAR.writeString(builder, plan);
+        return new Page(builder.build());
```

Comment:
The problem is that `Operator` must return a blocked future from `isBlocked()` if they can't make progress. As long as that future is done `Driver.processInternal` will call `getOutput` and the driver will be scheduled on a thread in `TaskExecutor`. This means that the operator ends up in a spin wait. This might not be *too* bad since there's a single `ExplainAnalyzeOperator` in the plan. However, it has to be planned in the coordinator and we don't have extra CPU in the coordinator.
This means that we need to implement `isBlocked`, but this requires a notification when all task stats become available. It's certainly possible to implement this (as will be done as part of @nileema's work), but right now there's only one interface on `QueryManager` to observe state changes (`waitForStateChange`) and that only works at the query level, not the task level